File: backend/src/routes/signup.ts
// Importar dependencias
import { Router, Request, Response } from 'express';
import bcrypt from 'bcrypt';
import dotenv from 'dotenv';

// Importar base de datos
import db from '../db';

// Cargar variables de entorno
dotenv.config();

// Definir las variables
const SALT_ROUNDS = 10;

// Crear el router
const router = Router();

/*
    RUTAS
*/

// Registro de usuario
router.post('/signup', async (req: Request, res: Response) => {
    let connection;
    try {
        const { username, mail, pwd, birthdate, gender } = req.body;
        connection = await db.promise().getConnection();
        
        // Comprobar si el usuario ya existe
        const [existingUser] = await connection.query('SELECT * FROM Usuario WHERE mail_usuario = ?', [mail]);
        if ((existingUser as any[]).length > 0) {
            res.status(409).send('El correo electrónico ya está en uso');
            return;
        }

        // Hashear la contraseña
        const pwdHash = await bcrypt.hash(pwd, SALT_ROUNDS);

        // Insertar el nuevo usuario en la base de datos
        await connection.query('INSERT INTO Usuario (nombre_usuario, mail_usuario, contrasenia, genero_usuario, rol, fecha_nacimiento) VALUES (?, ?, ?, ?, "Usuario", ?)', [username, mail, pwdHash, gender, birthdate]);
        
        // Enviar respuesta de éxito
        res.status(201).send('Usuario registrado con éxito');
    } catch (err) {
        console.error('Error al registrar el usuario:', err);
        res.status(500).send('Error interno del servidor');
    } finally {
        // Liberar la conexión a la base de datos
        if (connection) {
            connection.release();
        }
    }
});

// Exportar el router
export default router;

File: backend/routes/signup.py
# Importar dependencias
from flask import Blueprint, request, jsonify
from dotenv import load_dotenv
import bcrypt
import os

# Cargar variables de entorno
load_dotenv()

# Definir variables
SALT_ROUNDS = os.getenv("SALT_ROUNDS")

# Crear un objeto Blueprint para la ruta de registro
signup_bp = Blueprint('signup', __name__)

'''
    RUTAS
'''

# Registro de usuario
@signup_bp.route('/signup', methods=['POST'])
def signup():
    from app import mysql
    cur = None
    try:
        data = request.get_json()
        username = data.get('username')
        mail = data.get('mail')
        pwd = data.get('pwd')
        bdate = data.get('birthdate')
        gender = data.get('gender')

        # Obtener la conexion a la base de datos
        cur = mysql.connection.cursor()

        # Comprobar si el usuario ya existe
        cur.execute("SELECT * FROM Usuario WHERE mail_usuario = %s", (mail,))
        user = cur.fetchone()
        if user is not None:
            return jsonify({'message': 'El usuario ya existe'}), 409
        
        # Hashear la contraseña
        pwdHash = bcrypt.hashpw(pwd.encode('utf-8'), bcrypt.gensalt(int(SALT_ROUNDS))).decode('utf-8')

        # Insertar el nuevo usuario en la base de datos
        cur.execute("INSERT INTO Usuario (nombre_usuario, mail_usuario, contrasenia, genero_usuario, rol, fecha_nacimiento) VALUES (%s, %s, %s, %s, 'Usuario', %s)", (username, mail, pwdHash, gender, bdate,))
        mysql.connection.commit()

        # Respuesta de éxito
        return jsonify({'message': 'Usuario registrado exitosamente'}), 201
    except Exception as err:
        print(f"Error al registrar el usuario: {err}")
        return jsonify({'message': 'Error interno del servidor'}), 500
    finally:
        # Cerrar el cursor
        if cur: cur.close()
